app/utils/write_rego.py

```diff
@@ -2,54 +2,56 @@
 
 from app.config.config import settings
 from app.server.github import GitHubOperations
-
 from .build_rego_file import build_rego
 
-github = GitHubOperations(settings.GITHUB_URL)
-
 initiate_rule = "package httpapi.authz\nimport input\ndefault allow = false\n\n\n\n"
 
 
-def write_to_file(rule, operation: str = "write") -> dict:
-    """
-    Write the rego file to the local git repository
-    :param rule: rules
-    :return: response dict to show the status of the request
-    """
-    # Define file path
-    file_path = f"{github.local_repo_path}/auth.rego"
+class WriteRego:
+    def __init__(self, access_token: str) -> None:
+        self.access_token = access_token
+        self.github = GitHubOperations(settings.GITHUB_URL, access_token)
 
-    # Initialize repository
-    github.initialize()
+    def write_to_file(self, rule: dict, operation: str = "write") -> None:
+        """
+        Write the rego file to the local git repository
+        :param rule: rules
+        :param operation: write or update
+        :return: response dict to show the status of the request
+        """
+        # Define file path
+        file_path = f"{self.github.local_repo_path}/auth.rego"
 
-    rule = {key: value for key, value in rule.items()}
+        # Initialize repository
+        self.github.initialize()
 
-    # check if the file exists
-    if os.path.exists(file_path):
-        with open(file_path, "r") as file:
-            data = file.read()
-            old_rule = data if operation == "write" and data else initiate_rule
+        rule = {key: value for key, value in rule.items()}
 
-    else:
-        # write the file
-        old_rule = initiate_rule
-    result = old_rule + build_rego(rule["rules"])
+        # check if the file exists
+        if os.path.exists(file_path):
+            with open(file_path, "r") as file:
+                data = file.read()
+                old_rule = data if operation == "write" and data else initiate_rule
 
-    with open(file_path, "w") as file:
-        file.write(result)
-    # Update GitHub
-    github.push()
+        else:
+            # write the file
+            old_rule = initiate_rule
+        result = old_rule + build_rego(rule["rules"])
 
+        with open(file_path, "w") as file:
+            file.write(result)
+        # Update GitHub
+        self.github.push()
 
-def delete_policy_file() -> bool:
-    file_path = f"{github.local_repo_path}/auth.rego"
+    def delete_policy_file(self) -> bool:
+        file_path = f"{self.github.local_repo_path}/auth.rego"
 
-    if not os.path.isfile(file_path):
-        raise FileNotFoundError
+        if not os.path.isfile(file_path):
+            raise FileNotFoundError
```

Comment:
Won't this break the app, if the file not found?

Reply:
It'll return an exception.it won't break the app. I'll see how that can be handled later